nemo_curator/classifiers/aegis.py

```diff
@@ -0,0 +1,271 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+
+os.environ["RAPIDS_NO_INITIALIZE"] = "1"
+os.environ["DASK_DATAFRAME__QUERY_PLANNING"] = "False"
+from dataclasses import dataclass
+from functools import lru_cache
+from typing import List, Optional, Union
+
+import cudf
+import torch
+import torch.nn as nn
+from crossfit import op
+from crossfit.backend.torch.hf.model import HFModel
+from peft import PeftModel
+from transformers import AutoConfig, AutoModelForCausalLM, AutoTokenizer
+
+from nemo_curator.classifiers.base import DistributedDataClassifier
+from nemo_curator.datasets import DocumentDataset
+from nemo_curator.utils.aegis_utils import format_aegis
+
+
+@dataclass
+class AegisConfig:
+    peft_model_name_or_path: str
+    token: Optional[Union[str, bool]] = None
+    pretrained_model_name_or_path: str = "meta-llama/LlamaGuard-7b"
```

Comment:
Sorry for the delay, had to wait for access to https://huggingface.co/meta-llama/LlamaGuard-7b - do you think we should add a note for the user about requesting access?

Reply:
Yeah I forgot to do that. I added a note in the docstring of the class and in the constructor. I also catch and rethrow a more helpful error message now so debugging should be easier. Let me know what you think.